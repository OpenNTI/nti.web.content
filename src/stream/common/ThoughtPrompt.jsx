import React from 'react';
import PropTypes from 'prop-types';
import { scoped } from '@nti/lib-locale';

const DEFAULT_TEXT = {
	prompt: 'Write something...'
};

const t = scoped('web-content.stream.ThoughtPrompt', DEFAULT_TEXT);

export default function ThoughtPrompt ({ onClick }) {
	return (
		<div className="new-thought" onClick={onClick}>
			<div className="prompt">{t('prompt')}</div>
		</div>
	);
}

ThoughtPrompt.propTypes = {
	onClick: PropTypes.func
};
