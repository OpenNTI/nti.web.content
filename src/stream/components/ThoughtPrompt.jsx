import React from 'react';
import PropTypes from 'prop-types';
import { scoped } from '@nti/lib-locale';
import { LinkTo } from '@nti/web-routing';

const DEFAULT_TEXT = {
	prompt: 'Write something...',
	promptWithTitle: 'Write something in %(title) ...'
};

const t = scoped('web-content.stream.ThoughtPrompt', DEFAULT_TEXT);

export default function ThoughtPrompt ({ title }) {
	return (
		<LinkTo.Object context="thought">
			<div className="new-thought">
				<div className="prompt">{title ? t('promptWithTitle', { title }) : t('prompt')}</div>
			</div>
		</LinkTo.Object>
	);
}

ThoughtPrompt.propTypes = {
	title: PropTypes.string
};
