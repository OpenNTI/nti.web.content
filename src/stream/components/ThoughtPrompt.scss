@import "~@nti/style-common";

.new-thought {
	background-color: #eaedf1;
    height: 60px;
    padding: rem-calc(8px);
	margin-bottom: rem-calc(5px);

	.prompt {
		background-color: white;
		line-height: 45px;
		border-radius: rem-calc(5px);
		box-shadow: 0 0 0 1px #dfe1e5;
		padding: 0 rem-calc(15px);
		color: $tertiary-grey;
		font-style: italic;
		font-weight: 300;
		font-size: 0.875em;
		cursor: pointer;
	}
}
