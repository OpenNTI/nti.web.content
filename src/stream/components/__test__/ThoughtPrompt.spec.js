import React from 'react';
import renderer from 'react-test-renderer';

import ThoughtPrompt from '../ThoughtPrompt';

/* eslint-env jest */
describe('Thought Prompt', () => {
	test('Simple', () => {
		const tree = renderer.create(<ThoughtPrompt />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
