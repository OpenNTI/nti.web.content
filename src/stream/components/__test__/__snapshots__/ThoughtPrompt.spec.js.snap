// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thought Prompt Simple 1`] = `
<div
  className="new-thought"
>
  <div
    className="prompt"
  >
    Write something...
  </div>
</div>
`;
